src/olympia/shelves/serializers.py

```diff
@@ -50,6 +51,13 @@ def get_addons(self, obj):
             addons = AddonSearchView(request=request).data
             request.GET = tmp
             return addons
+        elif obj.endpoint == 'collections':
+            request = self.context.get('request', None)
```

Comment:
nit: you don't need to specify `None` as the default with a dict `.get` call in python - it's the default anyway.